modules/assim.sequential/R/sda.enkf.R

```diff
@@ -240,42 +255,58 @@ sda.enkf <- function(settings, IC, prior, obs.mean, obs.cov, variables,
     
     obs = !is.na(obs.mean[[t]])
     
-    ### ANALYSIS
-    if(any(obs)){ #if no obs skip analysis
+    ###-------------------------------------------------------------------###
+    ### analysis                                                          ###
+    ###-------------------------------------------------------------------###  
+    if(any(obs)){ #if no observations skip analysis
     mu.f = as.numeric(apply(X,2,mean,na.rm=TRUE))
     Pf   = cov(X)
-    Y    = obs.mean[[t]][[1]][pmatch(colnames(X), names(obs.mean[[t]][[1]]))]
+    Y    = obs.mean[[t]][pmatch(colnames(X), names(obs.mean[[t]]))]
 
-    H = diag(length(obs.mean[[t]][[1]]))
-    R = obs.cov[[t]]#diag(as.numeric(obs.sd[[t]][[1]][pmatch(colnames(X),names(obs.mean[[t]][[1]]))])^2)
+    R = as.matrix(obs.cov[[t]])#how do we make sure this is in the right order?
```

Comment:
Doesn't it have to be in the same order as obs.mean?  That seems like a safe assumption to me